### PR TITLE
Jasmine Type safe method names list on a generic for CreateSpyObj

### PR DESCRIPTION
**Pull request by jonyeezs**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
  * _Tried on Typescript playground_
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Using newer Typescript features to enhance the typing. No definition was altered.

### CODE REVIEW

**Comment by lukas-zech-software** [comment]
lgtm 👍

File: types/jasmine/index.d.ts [hunk 1]
```diff
@@ -146,7 +145,7 @@ declare namespace jasmine {
     function createSpy(name?: string, originalFn?: Function): Spy;
 
     function createSpyObj(baseName: string, methodNames: SpyObjMethodNames): any;
-    function createSpyObj<T>(baseName: string, methodNames: SpyObjMethodNames): SpyObj<T>;
+    function createSpyObj<T>(baseName: string, methodNames: SpyObjMethodNames<T>): SpyObj<T>;
 
     function createSpyObj(methodNames: SpyObjMethodNames): any;
     function createSpyObj<T>(methodNames: SpyObjMethodNames): SpyObj<T>;
```
Comment (AnthonyMacKinnon) [inline types/jasmine/index.d.ts]: Should the same change be applied here (i.e. `methodNames: SpyObjMethodNames<T>`)?

**Comment by jonyeezs** [comment]
That would allow an interface. That allowed methodName entries that aren't
key of T.

This should ensure it doesn't have a breaking change.

**Review by AnthonyMacKinnon — APPROVED** [review]

**Comment by cyungmann** [comment]
@jonyeezs why is `SpyObjMethodNames<T>` a union type that includes `string[]`? Doesn't that kind of defeat the type safety? How is this intended to be used?

**Comment by jonyeezs** [comment]
I agree. String[] was the old implementation. Not sure if people were using
that "unsafely". And didn't want to cause a breaking change. But you Are
right, we should enforce it. Would you like to do a PR for that?

**Pull request merged**